EIPS/eip-4762.md

```diff
@@ -186,13 +196,37 @@ If `value` is non-zero:
 (tx.target, 0, BASIC_DATA_LEAF_KEY)
 ```
 
+#### Setcode transactions
+
+For EIP-7702 transactions, for each `[[chain_id, address, nonce, y_parity, r, s], ...]` entry in the authorization list, process the additional access events:
+
+```
+(address, 0, BASIC_DATA_LEAF_KEY)
+(address, 0, CODEHASH_LEAF_KEY)
+(address, 0, CODE_OFFSET)
+```
+
+If the authorization is valid, process the following write events:
+
+```
+(address, 0, BASIC_DATA_LEAF_KEY)
+(address, 0, CODE_OFFSET)
+```
+
+If `address` is not present in the state, also process the write event:
+
+```
+(address, 0, CODEHASH_LEAF_KEY)
+```
+
 ### Witness gas costs
 
 Remove the following gas costs:
 
  * Increased gas cost of `CALL` if it is nonzero-value-sending
  * [EIP-2200](./eip-2200.md) `SSTORE` gas costs except for the `SLOAD_GAS`
  * 200 per byte contract code cost
+ * `PER_EMPTY_ACCOUNT_COST` and `PER_AUTH_BASE_COST`
```

Comment:
This is potentially a controversial move, but since we have the stateless costs to cover for the writes to the state, that should match. Currently, they are covered by the `21000` intrinsic gas. Are these extra costs related to the transaction itself, or the state being written?

 - If it's the state being written, then I think we should drop them in verkle
 - if it's to disincentivize growing the transaction payload, then this line needs to be removed.

Reply:
Ok

Reply:
also, its done is a different way and i dont understand the reason behind that
we charge the `PER_EMPTY_ACCOUNT_COST` always and then refund `PER_AUTH_BASE_COST` if the account exists
looking at how the gas costs are implemented in this EIP. I think we should keep the gas costs as it is and deal with them in the way we deal with intrisic gas costs for transactions right now